### ISSUE TITLE
block-extension-tooltip build error

### ISSUE THREAD

**Issue by moniika** [RaspberryPiFoundation/blockly-samples#805]
**Category**
Plugin

**Component**
block-extension-tooltip

**Describe the bug**
There is an error when building the production version of block-extension-tooltip.

**To Reproduce**

1. In the `block-extension-tooltip` directory, run `npm run build`
```
ERROR in /usr/local/google/home/kozbial/github/blockly/samples-moniika/plugins/block-extension-tooltip/test/index.ts
ERROR in /usr/local/google/home/kozbial/github/blockly/samples-moniika/plugins/block-extension-tooltip/test/index.ts(79,53):
TS2345: Argument of type '(blocklyDiv: HTMLElement, options: BlocklyOptions) => WorkspaceSvg' is not assignable to parameter of type '(blocklyDiv: HTMLElement, options: BlocklyOptions) => Workspace'.
  Types of parameters 'options' and 'options' are incompatible.
    Type 'import("/usr/local/google/home/kozbial/github/blockly/samples-moniika/plugins/dev-tools/node_modules/blockly/blockly").BlocklyOptions' is not assignable to type 'import("/usr/local/google/home/kozbial/github/blockly/samples-moniika/plugins/block-extension-tooltip/node_modules/blockly/blockly").BlocklyOptions'.
      Types of property 'parentWorkspace' are incompatible.
        Type 'WorkspaceSvg' is missing the following properties from type 'WorkspaceSvg': getMetricsManager, setMetricsManager, getCachedParentSvgSize, setCachedParentSvgSize, and 2 more.
```

**Expected behavior**
No build error.

**Additional context**
May be related (or possibly the cause of) the issue with the released version of the plugin missing it's compressed files (tracked in https://github.com/google/blockly-samples/issues/793)

**Comment by maribethb** [comment]
So you get errors like this if the versions of Blockly don't match in dev-tools and block-extension-tooltip.

The current status is dev-tools is on 7.20211209.0 and block-extension-tooltip is on 5.20210325.1 and you get this error, or similar ones ('block is not compatible with type block' or something similar that sounds like nonsense)

If you set them both to v7 or both to v5 then the build succeeds.

This is frustrating in cases like currently where block-extension-tooltip is not actually compatible with v7 and downgrading dev-tools is not an option.

But I don't know WHY this is the case. I don't know why the version that's in dev-tools is even coming into play at all. `dev-scripts` which runs the compilation doesn't depend on `dev-tools`. But I don't even know if this error is coming from tsc or from babel. It looks like the error is coming from the `test` directory of the plugin, which is where we depend on `dev-tools`, but also isn't something we actually need to publish... so maybe if we didn't compile the `test` directory that might work?

edit: the error is coming from tsc. If you run `npx tsc test/*.ts` then you will see these errors and it also confirms the conflicting files it mentions are in node_modules/blockly for this plugin and also for the one under dev-tools

If anyone out there is reading this and knows more about TypeScript than me we would love assistance in figuring this out.

**Comment by maribethb** [comment]
So I was previously getting stuff like:

```shell
Argument of type '(block: Blockly.BlockSvg) => HTMLDivElement' is not assignable to parameter of type 'TooltipRender'.
  Types of parameters 'block' and 'block' are incompatible.
    Type 'BlockSvg__Class' is missing the following properties from type 'BlockSvg': svgGroup_, renderIsInProgress_, useDragSurface_, eventsInit_, and 30 more.

17 registerTooltipExtension((block: Blockly.BlockSvg) => {
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../dev-tools/node_modules/blockly/msg/msg.d.ts:13:1 - error TS6200: Definitions of the following identifiers conflict with those in another file: ADD_COMMENT, CANNOT_DELETE_VARIABLE_PROCEDURE, CHANGE_VALUE_TITLE, CLEAN_UP, COLLAPSED_WARNINGS_WARNING, COLLAPSE_ALL, COLLAPSE_BLOCK, COLOUR_BLEND_COLOUR1, COLOUR_BLEND_COLOUR2, COLOUR_BLEND_HELPURL, COLOUR_BLEND_RATIO, COLOUR_BLEND_TITLE, COLOUR_BLEND_TOOLTIP, COLOUR_HUE, [many more identifiers listed here]
13 declare namespace Blockly.Msg {
   ~~~~~~~

  node_modules/blockly/msg/msg.d.ts:13:1
    13 declare namespace Blockly.Msg {
       ~~~~~~~
    Conflicts are in this file.
```


then I tried adding

```json
"baseUrl": ".",
    "paths": {"blockly": ["node_modules/blockly"]},
```

to the tsconfig and then running `npx tsc --project tsconfig.json` and now I ONLY get the

```shell
test/index.ts:17:26 - error TS2345: Argument of type '(block: Blockly.BlockSvg) => HTMLDivElement' is not assignable to parameter of type 'TooltipRender'.
  Types of parameters 'block' and 'block' are incompatible.
    Type 'BlockSvg__Class' is missing the following properties from type 'BlockSvg': svgGroup_, renderIsInProgress_, useDragSurface_, eventsInit_, and 30 more.

17 registerTooltipExtension((block: Blockly.BlockSvg) => {
```

error. That feels like progress but not the magical cure I was hoping for. My understanding of this combinations of flags is that it's me telling tsc where to go to look for type definitions, so I am explicitly telling it to look in the current directory for blockly instead of it happening to find the one in dev-tools. Not sure how correct that is. So now the error with `BlockSvg__Class` vs `BlockSvg` might be a different issue. WIll keep looking for a little while longer before giving up.

**Comment by maribethb** [comment]
Okay and the last error is because in `src` we import from `blockly/core` and in `test` we import from `blockly`. It doesn't like that. I don't understand exactly why but right now I don't care. I tried having `test` import from `blockly/core` which does resolve this problem but creates a nearly identical one because dev-tools imports only from `blockly` so we get the same problem with `createPlayground`. So to resolve this I just made `src` import `blockly` which increases the module size a bit and gives a warning... but at least it builds!
